lib/analyse.ml

```diff
@@ -27,51 +33,102 @@ module Analysis = struct
   let is_duniverse _ = false
 
   let ocamlformat_source _ = None
-
-  let of_dir ~head ~job dir =
+  
+  let check_opam_version =
+    let version_2 = OpamVersion.of_string "2" in
+    fun name opam ->
+      let opam_version = OpamFile.OPAM.opam_version opam in
+      if OpamVersion.compare opam_version version_2 < 0 then
+        Fmt.failwith "Package %S uses unsupported opam version %s (need >= 2)" name (OpamVersion.to_string opam_version)
+
+  let of_dir ~job ~master dir =
     (* TODO: Check if the PR added an opam file in packages/<pkg> instead of packages/<pkg>/<pkg>.<ver> (common mistake) *)
     (* TODO: Split modified vs. added (using git diff --name-status) *)
-    let (_, master) = Current_git.Commit.hash head in
-    let master = Option.get master in
-    let fmt = Printf.sprintf in
-    let cmd = "", [| "sh"; "-c"; fmt {|git diff %s | sed -E -n -e 's,^\+\+\+ b/packages/[^/]*/([^/]*)/.*,\1,p'|} master |] in
-    Current.Process.check_output ~cwd:dir ~cancellable:true ~job cmd >>!= fun output ->
-    let opam_files =
-      String.split_on_char '\n' output
-      |> List.filter (fun s -> not (String.equal s ""))
-      |> List.sort String.compare
-    in
-    let r = { opam_files } in
-    Current.Job.log job "@[<v2>Results:@,%a@]" Yojson.Safe.(pretty_print ~std:true) (to_yojson r);
-    Lwt.return (Ok r)
+    let master = Current_git.Commit.hash master in
+    let cmd = "", [| "git"; "merge"; master |] in
```

Comment:
Some pair of comment as in https://github.com/ocurrent/opam-repo-ci/pull/3/files#r438897884